### PR TITLE
Escape filter list in a way that can be copied and pasted into a shell with `NOMATCH`

### PR DESCRIPTION
**Pull request by bwrsandman**

### CODE REVIEW

File: post/clang_tidy_review/clang_tidy_review/__init__.py [hunk 1]
```diff
@@ -71,7 +72,9 @@ def build_clang_tidy_warnings(
 
     start = datetime.datetime.now()
     try:
-        with message_group(f"Running:\n\t{command}"):
+        with message_group(
+            f"Running:\n\t{' '.join(shlex.quote(arg) for arg in shlex.split(command))}"
```
Comment (bwrsandman) [inline post/clang_tidy_review/clang_tidy_review/__init__.py:76]: Might require `shlex.join`

**Comment by ZedThree** [comment]
Please could you give an example of before and after? I'm not quite sure what the issue is!

**Comment by bwrsandman** [comment]
Currently it will print this
```bash
clang-tidy -p=cmake-build-presets/ninja-multi-vcpkg --config-file=".clang-tidy" -line-filter=["{'name': 'src/Game.cpp', 'lines': [[677, 677], [712, 712]]}"] "src/Game.cpp" --export-fixes=clang_tidy_review.yaml/mnt 
```
This causes issues with some interpreters, I tried mac's bash and zsh
```
zsh: no matches found: -line-filter=[{'name': 'src/Game.cpp', 'lines': [[677, 677], [712, 712]]}]
```
Escaping line filter properly gives a slightly uglier command but you can the use it in your script.

**Comment by bwrsandman** [comment]
Still need to test this version of the fix

**Review by ZedThree — APPROVED** [review]
Thanks @bwrsandman, this looks great!

**Pull request merged**